src/bytes.rs

```diff
@@ -63,6 +79,16 @@ impl ByteBuf {
             bytes: Vec::with_capacity(cap)
         }
     }
+
+    pub fn as_vec(self) -> Vec<u8> {
+        self.bytes
+    }
+}
```

Comment:
I would suggest removing 
```rust
impl<T> From<T> for ByteBuf where T: Into<Vec<u8>>
```
in favor of simpler
```rust
impl From<Vec<u8>> for ByteBuf
```
Though that would be a breaking change.